### PR TITLE
Performance Optimization

### PR DESCRIPTION
**Pull request by AlbumenJ**
## What is the purpose of the change

Revert "Revert "feat: change default mode (https://github.com/apache/dubbo/pull/11653)" (https://github.com/apache/dubbo/pull/11973)"

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — COMMENTED** [review]
@aamingaa PTAL

File: dubbo-remoting/dubbo-remoting-netty/src/test/java/org/apache/dubbo/remoting/exchange/support/header/HeartbeatHandlerTest.java [hunk 1]
```diff
@@ -118,6 +131,11 @@ void testClientHeartbeat() throws Exception {
             .addParameter(Constants.EXCHANGER_KEY, HeaderExchanger.NAME)
             .addParameter(Constants.TRANSPORTER_KEY, "netty3")
             .addParameter(Constants.CODEC_KEY, "telnet");
+        ApplicationModel applicationModel = ApplicationModel.defaultModel();
```
Comment (aamingaa) [inline dubbo-remoting/dubbo-remoting-netty/src/test/java/org/apache/dubbo/remoting/exchange/support/header/HeartbeatHandlerTest.java:134]: duplicate with line 108 ～ 112？

File: dubbo-common/src/main/java/org/apache/dubbo/rpc/executor/AbstractIsolationExecutorSupport.java [hunk 2]
```diff
@@ -50,15 +50,17 @@ public Executor getExecutor(Object data) {
...
-                return executorRepository.getExecutor(serviceUrl);
+                return executorRepository.getExecutor(providerModel, serviceUrl);
             }
         }
-        return executorRepository.getExecutor(serviceUrls.get(0));
+        return executorRepository.getExecutor(providerModel, serviceUrls.get(0));
     }
 
-    protected abstract ServiceKey getServiceKey(Object data);
+    protected ServiceKey getServiceKey(Object data) {
```
Comment (aamingaa) [inline dubbo-common/src/main/java/org/apache/dubbo/rpc/executor/AbstractIsolationExecutorSupport.java:59]: why change this method from abstract ？

Comment (AlbumenJ) [reply dubbo-common/src/main/java/org/apache/dubbo/rpc/executor/AbstractIsolationExecutorSupport.java:59]: Dubbo Protocol do not need to override this method.

File: dubbo-remoting/dubbo-remoting-netty/src/test/java/org/apache/dubbo/remoting/exchange/support/header/HeartbeatHandlerTest.java (hunk 1 above)
Comment (AlbumenJ) [reply dubbo-remoting/dubbo-remoting-netty/src/test/java/org/apache/dubbo/remoting/exchange/support/header/HeartbeatHandlerTest.java:134]: For ut

File: dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/manager/DefaultExecutorRepository.java [hunk 3]
```diff
@@ -131,16 +134,36 @@ private String getExecutorSecondKey(URL url) {
...
+            return getProviderKey((ProviderModel) serviceModel, url);
+        }
+    }
+
     private String getConsumerKey(URL url) {
         // Consumer's executor is sharing globally, key=Integer.MAX_VALUE
         return String.valueOf(Integer.MAX_VALUE);
     }
 
+    private static final String MAX_KEY = String.valueOf(Integer.MAX_VALUE);
```
[the remark this answers is not in the document]
Comment (AlbumenJ) [reply dubbo-common/src/main/java/org/apache/dubbo/common/threadpool/manager/DefaultExecutorRepository.java:150]: done

**Review by EarthChen — APPROVED** [review]
LGTM

File: dubbo-common/src/main/java/org/apache/dubbo/rpc/model/FrameworkServiceRepository.java [hunk 4]
```diff
@@ -83,18 +78,10 @@ public List<ProviderModel> lookupExportedServicesWithoutGroup(String key) {
...
-    }
-
     public ProviderModel lookupExportedService(String serviceKey) {
         return providers.get(serviceKey);
     }
 
-    public List<URL> lookupRegisteredProviderUrlsWithoutGroup(String key) {
-        return providerUrlsWithoutGroup.get(key);
-    }
-
```
Comment (chickenlj) [inline dubbo-common/src/main/java/org/apache/dubbo/rpc/model/FrameworkServiceRepository.java:97]: Why are all these public APIs removed?

**Pull request merged**